### ISSUE TITLE
setUserNotificationsEnabled and isUserNotificationsEnabled does not match permissions

### ISSUE THREAD

**Issue by ddennis** [urbanairship/phonegap-ua-push#124]
I use Xcode 7.3.1 running on iphone 6+ version 9.2.1, everything is called after the ***device ready event***.

I have a production version i can install via testflight and a development version i run through xcode. Push is working fine.

My workflow while "testing" through xcode is to uninstall the app, build from xcode which launches the app and then click button to ask for permission. allow "push notifications" - Check output in console.... and repeat. 

I am asking for push permissions when user clicks a button in my UI. When the button is clicked i  call the setUserNotificationsEnabled() function like so:

    UAirship.setUserNotificationsEnabled(true, function (enabled) {     
        console.log(enabled)
    }) 

The enabled parameter returned in the callback is always ***null***

Further more, when clicking the button, the call triggers the native popup which prompts the user if they want to recieve push notifications. I am expecting the callback function would be called when the user clicks ***allow/don't allow***, but instead the callback is called right away with the value ***null***

## If i try to validate if the user has allowed notifications:

    UAirship.isUserNotificationsEnabled(function(isEnabled) {
          console.log( isEnabled)
    })

This function always returnes 1 , no matter if the user has allowed or refused to recieved notifications.

Any help would be much appreciated.
/Dennis

**Comment by rlepinski** [comment]
The SDK's user notifications enabled does not reflect if the user accepted/declined user notifications. As soon as you enable user notifications in the SDK, it will prompt the user to accept them.  Each method takes 2 callbacks for success and failure. We did this across all methods for consistency. They were more to provide if the method actually is supported on the device not for result.

A solution would be to expose a new method, something like isOptedIn. On iOS it would be:

```
- (void)isOptedIn:(CDVInvokedUrlCommand *)command {
    [self performCallbackWithCommand:command withBlock:^(NSArray *args, UACordovaCompletionHandler completionHandler) {
        BOOL optedIn = [UAirship push].currentEnabledNotificationTypes != 0;
        completionHandler(CDVCommandStatus_OK, [NSNumber numberWithBool:optedIn]);
    }];
}
```

I can add it to our teams backlog to support. If you would like to do a PR to add support for it earlier, feel free. I can provide any guidance that you would need.

**Comment by ddennis** [comment]
Thank you for taking the time to explain why it's not working as expected. 

From a users point of view, i would expect the plugin to have this functionality, especially when urban-airship is a **paid service**, if i want to use it in production. But i understand that native IOS might be your first priority and not a cordova plugin.

I will try your solution, but my experience with developing cordova plugings is very limited and my knowledge about IOS is **none** Therefor it would be very helpfull if you could add this to your backlog.

For me personally i am looking for a plug and play solution, It seems the [cordova local notifications](https://github.com/katzer/cordova-plugin-local-notifications) already has this feature.

Thanks again.

**Comment by rlepinski** [comment]
@ddennis The request is on our backlog, and we hope to get to it by next week.

**Comment by rlepinski** [comment]
@ddennis https://github.com/urbanairship/phonegap-ua-push/pull/126. Once its reviewed and tested we will release an update.

**Comment by rlepinski** [comment]
Released in 5.4.0
